pkg/doc/signature/verifier/verifier.go

```diff
@@ -23,17 +23,34 @@ type signatureSuite interface {
 	GetDigest(doc []byte) []byte
 
 	// Verify will verify signature against public key
-	Verify(pubKey []byte, doc []byte, signature []byte) error
+	Verify(pubKey *PublicKey, doc []byte, signature []byte) error
 
 	// Accept registers this signature suite with the given signature type
 	Accept(signatureType string) bool
 }
 
+// PublicKey contains a result of public key resolution.
+type PublicKey struct {
+	Type  PublicKeyType
+	Value []byte
+}
+
+// PublicKeyType is a type of a public key
+type PublicKeyType string
+
+const (
+	// RSAPublicKeyType public key type
+	RSAPublicKeyType = PublicKeyType("RSA")
+
+	// Ed25519PublicKeyType public key type
+	Ed25519PublicKeyType = PublicKeyType(kms.Ed25519Type)
+)
```

Comment:
Instead of creating new key type struct here, add `RSA` key type in kms/api.go and reference its KeyTypes directly.

We should have only one keyType struct across the framework to avoid duplication.

Reply:
OK, I wil change that!

Reply:
changed